sdk/provisioning/Azure.Provisioning.CloudMachine/src/CloudMachineInfrastructure.cs

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Provisioning.Authorization;
+using Azure.Provisioning.EventGrid;
+using Azure.Provisioning.Expressions;
+using Azure.Provisioning.Resources;
+using Azure.Provisioning.Roles;
+using Azure.Provisioning.ServiceBus;
+using Azure.Provisioning.Storage;
+
+namespace Azure.Provisioning.CloudMachine;
+
+public class CloudMachineInfrastructure : Infrastructure
+{
+    private readonly string _name;
+    private UserAssignedIdentity _identity;
+    private StorageAccount _storage;
+    private BlobService _blobs;
+    private BlobContainer _container;
+    private ServiceBusNamespace _serviceBusNamespace;
+    private ServiceBusNamespaceAuthorizationRule _serviceBusNamespaceAuthorizationRule;
+    private ServiceBusTopic _serviceBusTopic_main;
+    private ServiceBusTopic _serviceBusTopic_app;
+    private ServiceBusSubscription _serviceBusSubscription_main;
+    private ServiceBusSubscription _serviceBusSubscription_app;
+    private SystemTopic _eventGridTopic_Blobs;
+    private SystemTopicEventSubscription _systemTopicEventSubscription;
+
+    /// <summary>
+    /// The common principalId parameter.
+    /// </summary>
+    public BicepParameter PrincipalIdParameter => new BicepParameter("principalId", typeof(string));
```

Comment:
Why are these public while everything else is private?  Just a point in time thing or ...?

Reply:
Point in time mostly - I haven't decided how to expose the other resources yet.